build-system/tasks/coverage-map.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const argv = require('minimist')(process.argv.slice(2));
+const fs = require('fs');
+const log = require('fancy-log');
+const puppeteer = require('puppeteer');
+const {exec} = require('../common/exec');
+
+const coverageJsonName = argv.json || 'out.json';
+const testUrl =
+  argv.url || 'http://localhost:8000/examples/everything.amp.html';
+
+async function compile() {
+  await exec(`gulp dist`);
+  // The question: How to cd to 'amphtml/dist' from anywhere in 'amphtml'?
+  await exec(`gulp serve --compiled`);
+}
+
+async function test() {
+  log('Opening browser and navigating to everything.amp.html...');
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+
+  // Enable JavaScript coverage
+  await Promise.all([page.coverage.startJSCoverage()]);
+  // Navigate to page
+  await page.goto(testUrl);
+  await page.setViewport({
+    width: 1200,
+    height: 800,
+  });
+  log('Scrolling to the end of the page...');
+  await autoScroll(page);
+  log('Testing completed.');
+  const [jsCoverage] = await Promise.all([page.coverage.stopJSCoverage()]);
+  const data = JSON.stringify(jsCoverage);
+  log('Writing to test.json in working directory...');
+  fs.writeFileSync(coverageJsonName, data);
```

Comment:
use the async version when possible since you're already using async api's mostly in the function

Reply:
I found a `fs.writeFile`. There doesn't seem to be an async version of `stringify`, though. Maybe I could wrap this in a `Promise.resolve()`? What do you think?

Reply:
you'll want to use `fs-extra` module instead of just plain `fs`. you can find some of its async functions api list in https://github.com/jprichardson/node-fs-extra

Reply:
Why use Promise.all above when there's just one promise to await?

Reply:
@rcebulko i believe that is removed already. this diff you're seeing here is outdated

Reply:
@erwinmombay The options in `fs-extra` doesn't work - I've tested it, and it generates a JSON that source map explorer will not accept. Doing `fs.writeFile()` works though - and it's asynchronous according to the documentation.
Please let me know what you think of this before I resolve this conversation.

Reply:
Per @rcebulko suggestion, I changed the import statement to `require('fs).promises` so the `writeFile` is now async.